test/Microsoft.AspNetCore.Server.KestrelTests/BadHttpRequestTests.cs

```diff
@@ -1,72 +1,99 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Threading.Tasks;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Server.KestrelTests
 {
     public class BadHttpRequestTests
     {
-        [Theory(Skip = "This test fails intermittently and needs to be investigated.")]
-        [InlineData("/ HTTP/1.1\r\n\r\n")]
-        [InlineData(" / HTTP/1.1\r\n\r\n")]
-        [InlineData("  / HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  / HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  /  HTTP/1.1\r\n\r\n")]
-        [InlineData("GET  HTTP/1.1\r\n\r\n")]
+        // Don't send more data than necessary to fail, otherwise the test might throw if
+        // it's still trying to send characters when the server had already shut down the connection
```

Comment:
Turns out it was not caused by the server shutting down the socket in both directions, since libuv shuts down the socket with `SD_SEND`.

The original issue was happening because Kestrel closed the connection while the client was still trying to write request bytes. This caused an exception to be thrown, which I though could be just swallowed so I could move on to trying to read the 400 response.

The problem seems to be that once we try to send data on a closed connection, winsock's (this only happens on Windows) error handling frees *everything* used by the client socket, *including* the buffered response from the server. So the test failed with an exception when we tried to read the response because it was not there anymore.

@mikeharder found an explanation here:

https://www.chilkatsoft.com/p/p_299.asp